src/main/java/de/azapps/kafkabackup/common/offset/OffsetSink.java

```diff
@@ -78,30 +78,14 @@ public void syncOffsets() {
     }
 
     private void syncOffsetsForGroup(String consumerGroup) throws IOException {
-        Map<TopicPartition, OffsetAndMetadata> partitionOffsetsAndMetadata = new HashMap<>();
-        partitionsReadLock.lock();
+        Map<TopicPartition, OffsetAndMetadata> partitionOffsetsAndMetadata;
+
         try {
-            // Due to https://issues.apache.org/jira/browse/KAFKA-9507, we need to fetch offsets one TopicPartition at a time
-            // and catch the exception that occurs if the consumer group is missing from the TopicPartition's committed offsets.
-            // TODO: fix when 2.4.1 is released!
-            for (TopicPartition tp : partitions) {
-                ListConsumerGroupOffsetsOptions listConsumerGroupOffsetsOptions = new ListConsumerGroupOffsetsOptions();
-                listConsumerGroupOffsetsOptions.topicPartitions(Collections.singletonList(tp));
```

Comment:
Most would be the same, but this line would be

```
listConsumerGroupOffsetsOptions.topicPartitions(partitions);
```
instead of the singleton list.